os/drivers/seclink/seclink_drv_key.c

```diff
@@ -29,71 +29,80 @@
 #endif
 #define SLDRV_TAG "[SECLINK_DRV_KEY]"
 
-#define NKEY_SLOT 4
+#define NKEY_SLOT 64
+/*
+ * +-----------------+------------------+
+ * |    ROM (0~31)   | RAM(32~63)       |
+ * +-----------------+------------------+
+ */
+#define KEY_ROM_ENTRY 0
+#define KEY_RAM_ENTRY 32
 static uint8_t g_key_inuse[NKEY_SLOT] = {0,};
+#define KEYSLOT_USE 0
+#define KEYSLOT_NOTUSE 1
 
-static int _set_key_slot(uint8_t num)
+static inline int _is_rom_area(uint8_t num)
 {
-	if (num >= NKEY_SLOT) {
-		return -1;
+	if (num < KEY_RAM_ENTRY) {
+		return 1;
 	}
-	g_key_inuse[num] = 1;
 	return 0;
 }
 
 
-static int _unset_key_slot(uint8_t num)
+static inline int _set_key_slot(uint8_t num)
 {
 	if (num >= NKEY_SLOT) {
 		return -1;
 	}
-	g_key_inuse[num] = 0;
+	g_key_inuse[num] = KEYSLOT_USE;
 	return 0;
 }
 
 
-static int _is_empty_key_slot(uint8_t num)
+static inline int _unset_key_slot(uint8_t num)
 {
 	if (num >= NKEY_SLOT) {
 		return -1;
 	}
-
-	return g_key_inuse[num] == 0 ? 1 : 0;
+	g_key_inuse[num] = KEYSLOT_NOTUSE;
+	return 0;
 }
 
 
-void hd_initialize_key_slot(void)
+static inline int _is_empty_key_slot(uint8_t num)
```

Comment:
'_is_' is usually used for boolean type.
I think _get_key_slot_state is correct, otherwise this should return 1 if it is empty slot.